### PR TITLE
Add attribute and history utils, fix ensembles issues

### LINKED ISSUE #402: Ensemble creation with files having 'end of frequency' dates
When creating an ensemble object on simulations that were calculated using frequncies such as 'Y' (vs 'YS') or 'M' (vs 'MS'). Ensemble creation usually succeeds but creates an object with varying time coordinates depending on the calendar of the input simulaitons. We can end up with an object with 2-3 times as many time coordinates as desired.  E.g. Jan 30 (from 360 day) and Jan31 from other calendars.  Resulting ensmeble data-arrays are padded with nans for simulations that do not have the same calendars (or end of month values)

I'm not sure the best way to deal with this would be but it could be something like  :

-  For each dataset : implement a check on the ds.time.dt.day values to ensure that they are a constant value. 

- Implement a 'uniform_day_of_month' option.   e.g. ``` uniform_day_of_month == 15```  would set the day value to the provided constant in ```ens_checktimes```  instead of using ``ds.time.dt.day``

E.g. adding an ``if`` in this section of code to modify the day value of the creation of the common datestamp
```
      if hasattr(ds, "time"):
            ds["time"] = xr.decode_cf(ds).time
            time_flag = True

            # get times - use common
            time1 = pd.to_datetime(
                {
                    "year": ds.time.dt.year,
                    "month": ds.time.dt.month,
                    "day": ds.time.dt.day,
                }
            )

 ````
Open to suggestions for dealing with it

### LINKED ISSUE #404: ensembles.ensemble_percentiles retains 'realization' dimension
* xclim version: 0.15.0
* Python version: 3.7
* Operating System: Linux

### Description
output dataset of ```ensembles.ensemble_percentiles()``` retains an unused "realization" dimension. Not a huge issue when working in xarray or netcdf but export to .csv via ds.to_Dataframe.to_csv causes n = len(ds.realization) copies of the percentiles data to be exported to the output .csv

adding a ```ds.drop('realization')```  to  ```ensembles.ensemble_percentiles()``` before the return should fix the issue
```

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #402 and fixes #404
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (minor / major / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 
As always I started fixing a small issue and fell down the rabbit hole...

1) New methods `xclim.core.formatting.merge_attributes` and `xclim.core.formatting.update_history` to handle the update of attributes when creating new dataset/dataarrays. Its pretty much the same that was done in the indicator class, but in independent reusable methods.

2) Fix of ensembles issues #402 and #404. 
a) Both `ensembles.ensemble_percentiles` and `ensembles.ensemble_mean_std_max_min` have been fix so that the output doesn't have the "realization" coordinate. Also, on the side, I used the methods added in 1 to update the `history` attribute of the output.

b) Added a `resample_freq` argument to  `ensembles.create_ensemble`. It allows (in `ensembles._ens_align_datasets`) the alignment of the different members if they have the same frequency, but not the same offset (ex: `M` vs. `MS`). The user needs to pass the frequency they want on the output.

b*) By the way, I removed the `ensembles._check_times` method, I think previous changes to `create_ensemble` made it useless. Because, it was not doing anything `xr.concat` doesn't do already. But if @tlogan2000 could test with some edge cases, we might not be covering everyting in the test suite.

b**) I added a line to make `create_ensemble` accept DataArrays. It simply transforms them in to dataset. It was just useful for me at testing time, and I don't see why not.

* **Does this PR introduce a breaking change?** 

Not that I can see.

### CODE REVIEW

File: xclim/core/indicator.py [hunk 1]
```diff
@@ -167,30 +159,33 @@ def __call__(self, *args, **kwds):
...
 
-        # Assume the first arguments are always the DataArray.
-        das = tuple(ba.arguments.pop(self._parameters[i]) for i in range(self._nvar))
-
         # Pre-computation validation checks
-        for da in das:
+        for da in das.values():
             self.validate(da)
-        self.cfprobe(*das)
+        self.cfprobe(*das.values())
```
[the remark this answers is not in the document]
Comment (aulemahal) [reply xclim/core/indicator.py]: Here `das` is a dictionary of DataArrays. It was a list before, but I wanted to keep track of the name, so I made it a dict. So `values` is called on the dict, not on the DataArray!

File: xclim/ensembles.py [hunk 2]
```diff
@@ -326,39 +275,33 @@ def _ens_align_datasets(
...
+                counts = time.resample(time=resample_freq).count()
+                if any(counts > 1):
+                    raise ValueError(
+                        f"Alignment of dataset #{i:02d} failed : its time axis cannot be resampled to freq {resample_freq}."
+                    )
+                time = counts.time
 
-            cal1 = xr.decode_cf(ds).time
-            ds.drop_vars("time")
             ds["time"] = pd.to_datetime(
```
[the remark this answers is not in the document]
Comment (aulemahal) [reply xclim/ensembles.py]: No, the more I dig, the more I think the bug in #406 is more a xarray bug than a deprecation.

[the remark this answers is not in the document]
Comment (aulemahal) [reply xclim/ensembles.py]: Here we only assign. The bug was coming from the use of `-=`.
